src/main/java/com/ss/lms/controller/BorrowerController.java

```diff
@@ -33,152 +31,98 @@
 import com.ss.lms.service.BorrowerService;
 import com.ss.lms.service.ConnectionUtil;
 
-
-
 @RestController
 public class BorrowerController {
-	
+
 	@Autowired
 	BorrowerService borrowerService;
-	
+
 	@Autowired
-	Branch branch;
-	
-	
+	Optional<Branch> branch;
 
 	@RequestMapping(path = "borrower/{cardNo}/library/{branchId}/books/{bookId}:checkout", method = RequestMethod.POST, produces = {
-					MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+			MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
 	public ResponseEntity<Loans> checkOutBook(@PathVariable("cardNo") Integer cardNo,
-            @PathVariable("branchId") Integer branchId,
-            @PathVariable("bookId") Integer bookId) throws SQLException {
-		
-		Borrower borrower =null;
-		Book book = null;
-		Branch branch =null;
-		
-	
+			@PathVariable("branchId") Integer branchId, @PathVariable("bookId") Integer bookId) throws SQLException {
+
+		Optional<Borrower> borrower = null;
+		Optional<Book> book = null;
+		Optional<Branch> branch = null;
+
 		try {
-			
-			try {
-			
-			borrower = borrowerService.readByCardNoEssentialData (cardNo);
+
+			borrower = borrowerService.readBorrowerByCardNo(cardNo);
 			book = borrowerService.readBooksById(bookId);
 			branch = borrowerService.readBranchsById(branchId);
-			
-			Copies copy = borrowerService.readCopyByBranchIDBookID(book, branch);
-			Integer noOfCopies = copy.getNoOfCopies();
-			
-			if (noOfCopies == 0) {
+
+			if (!(borrower.isPresent() && book.isPresent() && branch.isPresent())) {
```

Comment:
i wouldnt change it at this point if it works. But another way to handle the optional output is to use the .get() method and catch the NoSuchElementException when it doesnt exist.